tests/unit_tests/arelle/test_validatexbrlcalcs.py

```diff
@@ -0,0 +1,140 @@
+from __future__ import annotations
+import pytest
+from _decimal import Decimal, InvalidOperation
+
+from arelle.ValidateXbrlCalcs import insignificantDigits
+
+
+@pytest.mark.parametrize('value, precision, decimals, scale, result, error', [
+    # precision
+    (1234, -1, None, None, ('0', '1234'), None),
+    (1234, 0, None, None, None, None),
+    (1234, 1, None, None, ('1000', '234'), None),
+    (1234, 2, None, None, ('1200', '34'), None),
+    (1234, 3, None, None, ('1230', '4'), None),
+    (1234, 4, None, None, None, None),
+    (1234, 5, None, None, None, None),
+
+    (1234.5678, -2, None, None, ('0', '1234.567800000000033833202906'), None),
+    (1234.5678, -1, None, None, ('0', '1234.567800000000033833202906'), None),
+    (1234.5678, 0, None, None, None, None),
+    (1234.5678, 1, None, None, ('1000', '234.567800000000033833202906'), None),
+    (1234.5678, 5, None, None, ('1234.5', '0.067800000000033833202906'), None),
+    (1234.5678, 9, None, None, ('1234.5678', '0.000000000000033833202906'), None),
+
+    # decimals
+    (1234, None, -5, None, ('0', '1234'), None),
+    (1234, None, -1, None, ('1230', '4'), None),
+    (1234, None, 0, None, None, None),
+    (1234, None, 1, None, None, None),
+
+    (1234.5678, None, -5, None, ('0', '1234.567800000000033833202906'), None),
+    (1234.5678, None, -1, None, ('1230', '4.567800000000033833202906'), None),
+    (1234.5678, None, 0, None, ('1234', '0.567800000000033833202906'), None),
+    (1234.5678, None, 1, None, ('1234.5', '0.067800000000033833202906'), None),
+    (1234.5678, None, 5, None, ('1234.5678', '0.000000000000033833202906'), None),
+
+    # precision + scale
```

Comment:
Neither `scale` or `precision` is used in core Arelle/plugins. But as mentioned in my other comment, I'm more concerned about establishing expected behavior so that we can verify that an upcoming fix for the `InvalidOperation` bug doesn't have unintended changes. It would likely also prove helpful if we decide to change how we handle the `scale` or `precision` arguments.